From e0ef77534af442d1ccd9bed2785acebe0d1eeed5 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Mon, 10 Jul 2023 13:55:48 +0200
Subject: [PATCH 0395/1141] dt-bindings: memory-controller: fix OMI yaml

Fix OMI yaml bindings (issues seen with dt_binding_check)

Change-Id: I28b9b1ab310cf74d3baf0bc635c91098162147a4
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/317429
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../memory-controllers/st,stm32-omi.yaml      | 96 ++++++++++---------
 1 file changed, 49 insertions(+), 47 deletions(-)

--- a/Documentation/devicetree/bindings/memory-controllers/st,stm32-omi.yaml
+++ b/Documentation/devicetree/bindings/memory-controllers/st,stm32-omi.yaml
@@ -4,7 +4,7 @@
 $id: http://devicetree.org/schemas/memory-controllers/st,stm32-omi.yaml#
 $schema: http://devicetree.org/meta-schemas/core.yaml#
 
-title: STM32 OSPI Memory Interface (OMI) bindings
+title: STM32 OSPI Memory Interface (OMI)
 
 maintainers:
   - Patrice Chotard <patrice.chotard@foss.st.com>
@@ -28,10 +28,10 @@ properties:
   reg:
     description: registers
 
-  '#address-cells':
+  "#address-cells":
     enum: [1, 2]
 
-  '#size-cells':
+  "#size-cells":
     enum: [0, 1]
 
   memory-region:
@@ -79,8 +79,8 @@ properties:
 required:
   - compatible
   - reg
-  - '#address-cells'
-  - '#size-cells'
+  - "#address-cells"
+  - "#size-cells"
   - clocks
   - interrupts
   - st,syscfg-dlyb
@@ -99,45 +99,46 @@ patternProperties:
               - jedec,spi-nor
               - spi-nand
 
-      st,max-frequency:
-        $ref: /schemas/types.yaml#/definitions/uint32
-        description:
-          Maximum clocking speed of the device in Hz.
-
-      st,tacc-ns:
-        description:
-          Device initial access time.
-
-      st,wzl:
-        $ref: /schemas/types.yaml#/definitions/string
-        description:
-          Enable write zero latency
-
-anyOf:
-  - if:
-      properties:
-        compatible:
-          enum:
-            - const: jedec,spi-nor
-            - const: spi-nand
-
-    then:
-      $ref: "/schemas/spi/spi-controller.yaml#"
-
-  - if:
-      properties:
-        compatible:
-          contains:
-            enum:
-              - cfi-flash
-
-    then:
-      $ref: "/schemas/mtd/mtd-physmap.yaml#"
-
-      required:
-        - st,max-frequency
-        - st,tacc-ns
-        - st,wzl
+    allOf:
+      - if:
+          properties:
+            compatible:
+              enum:
+                - jedec,spi-nor
+                - spi-nand
+
+        then:
+          $ref: "/schemas/spi/spi-controller.yaml#"
+
+      - if:
+          properties:
+            compatible:
+              contains:
+                enum:
+                  - cfi-flash
+
+        then:
+          $ref: "/schemas/mtd/mtd-physmap.yaml#"
+
+          properties:
+            st,max-frequency:
+              $ref: /schemas/types.yaml#/definitions/uint32
+              description:
+                Maximum clocking speed of the device in Hz.
+
+            st,tacc-ns:
+              description:
+                Device initial access time.
+
+            st,wzl:
+              $ref: /schemas/types.yaml#/definitions/flag
+              description:
+                Enable write zero latency
+
+          required:
+            - st,max-frequency
+            - st,tacc-ns
+            - st,wzl
 
 additionalProperties: false
 
@@ -152,8 +153,8 @@ examples:
       reg = <0x40430000 0x400>;
       memory-region = <&mm_ospi1>;
       interrupts = <GIC_SPI 163 IRQ_TYPE_LEVEL_HIGH>;
-      dmas = <&hpdma 2 0x22 0x00003121 0x0>,
-             <&hpdma 2 0x22 0x00003112 0x0>;
+      dmas = <&hpdma 2 0x62 0x00003121 0x0>,
+             <&hpdma 2 0x42 0x00003112 0x0>;
       dma-names = "tx", "rx";
       st,syscfg-dlyb = <&syscfg 0x1000>;
       clocks = <&scmi_clk CK_SCMI_OSPI1>;
@@ -181,7 +182,7 @@ examples:
       reg = <0x40430000 0x400>;
       memory-region = <&mm_ospi1>;
       interrupts = <GIC_SPI 163 IRQ_TYPE_LEVEL_HIGH>;
-      dmas = <&hpdma 2 0x63 0x00003011 0x0>;
+      dmas = <&hpdma 2 0x22 0x00003011 0x0>;
       dma-names = "rx";
       st,syscfg-dlyb = <&syscfg 0x1000>;
       clocks = <&scmi_clk CK_SCMI_OSPI1>;
@@ -196,5 +197,6 @@ examples:
         reg = <0x2 0x0 0x4000000>;
         st,max-frequency = <100000000>;
         st,tacc-ns = <48>;
+        st,wzl;
       };
     };
